sklearn/decomposition/pca.py

```diff
@@ -260,52 +321,71 @@ def fit_transform(self, X, y=None):
         return U
 
     def _fit(self, X):
-        """Fit the model on X
+        """Dispatch to the right submethod depending on the chosen solver."""
+        X = check_array(X, dtype=[np.float64], ensure_2d=True,
+                        copy=self.copy)
 
-        Parameters
-        ----------
-        X: array-like, shape (n_samples, n_features)
-            Training vector, where n_samples in the number of samples and
-            n_features is the number of features.
+        # Handle n_components==None
+        if self.n_components is None:
+            n_components = X.shape[1]
+        else:
+            n_components = self.n_components
 
-        Returns
-        -------
-        U, s, V : ndarrays
-            The SVD of the input data, copied and centered when
-            requested.
-        """
-        X = check_array(X)
+        # Handle svd_solver
+        svd_solver = self.svd_solver
+        if svd_solver == 'auto':
+            # Small problem, just call full PCA
+            if max(X.shape) <= 500:
```

Comment:
I'm sorry if I missed this is in the long discussion, but is this claim supported by benchmarks? Would it still be useful to use `randomized` if `n_components` is say 2 and `max(X.shape)` is 500.